packages/debug/index.js

```diff
@@ -9,4 +9,19 @@ module.exports = Object.assign({}, addonBaseConfig, {
   externalDependenciesForPrivateModule() {
     return [];
   },
+  isEnabled() {
+    const { hasOwnProperty } = Object.prototype;
+    const config = this.parent.config(this.app.env);
+    const userDefinedProperty = hasOwnProperty.call(config, 'includeDebug');
+    if (userDefinedProperty) {
+      // use the user defined property from environment.js
+      return config.includeDebug;
```

Comment:
I'd default to using an `option` for this, e.g. in `ember-cli-build` someone would specify:

```js
new EmberApp(defaults, {
  emberData: {
    includeDataAdapterInProduction: false
  }
})
```

Then we would check options for this in `index.js` of `debug` package:

```js
config() {
    let optionFlag =
      this.app &&
      this.app.options &&
      this.app.options.emberData &&
      this.app.options.emberData.includeDataAdapterInProduction;

    return {
      emberData: {
        enableRecordDataRFCBuild: optionFlag !== undefined ? optionFlag : false,
      },
    };
  }
```

This also gives the `ember-data` package a way to alter the default by setting this option:

```js
setupOptions() {
  if (this.app) {
    let options = this.app.options = this.app.options || {};
    options.emberData = options.emberData || {};
    if (options.emberData.includeDataAdapterInProduction === undefined) {
       options.emberData.includeDataAdapterInProduction = true;
    } 
  }
```

`ember-data` would need to call this during the included hook I believe, which will be called prior to the `isEnabled` hook in the `debug` package requesting the config.

Reply:
This sounds good, but I'm confused about how the ember-cli-build options actually works. The comments in the file https://github.com/emberjs/data/blob/master/packages/-ember-data/ember-cli-build.js#L18 say it is only for the dummy app...I also tried putting code in there but never found the right command or location to run the commands to have it run...I'll keep playing around.